demos/dialog/index.js

```diff
@@ -34,3 +36,61 @@ window.toggleActions.onclick = function() {
   dialog.hideActions = hideAction;
   hideActionSpan.innerText = hideAction;
 };
+
+class DialogWithFormValidation extends LitElement {  
+  static get properties() {
+    return {
+      isDisabled: {type: Boolean},
+      open: {type: Boolean},
+    };
+  }
+
+  constructor() {
+    super();
+    this.isDisabled = true;
+    this.open = false;
+  }
+
+  onInputListener(e) {
+    this.isDisabled = !e.target.checkValidity();
+  }
+
+  onClosingListener(e) {
+    this.open = false
+    // we could call APIs here ...
+  }
+
+  render() {
+    return html`
+      <mwc-dialog
+        @closing="${this.onClosingListener}"
+        heading="Form Validation"
+        .open="${this.open}">
+        <div>
+          Our primary action button is disabled
+          until our textfield has valid input.
+        </div>
+        <mwc-textfield
+            dialogInitialFocus
+            @input="${this.onInputListener}"
+            label="Name"
+            pattern="^[a-zA-Z]{3,}$"
+            placeholder="Casey"
+            required
+            type="text">
+        </mwc-textfield>
+        <mwc-button
+          dialogAction="close"
```

Comment:
I think the pattern we want to show off is to avoid dialogAction with form validation and to do it manually, right? I don't think it'd be necessary to disable/enable the button.

Instead, on clicking the button, you can check the text field's validity and then choose whether or not to manually close the dialog.

Reply:
I see!
Just to clarify, what do we mean by "manually close the dialog" ? Do we want to close the dialog once the contents of a textfield is valid programmatically? or simply don't let our users close the close until text field has valid input? :)

Reply:
I would say close the dialog rather than disable the button to prevent closing.

The common use case we get is a request for "how do I stop or cancel a dialogAction?" Usually the problem is the developer doesn't know whether or not the value is valid by the time they click the button to close the dialog, such as waiting on a server response.

```js
document.querySelector('.close-button').addEventListener('click', () => {
  const valid = checkValid(); // validate, possible asynchronous such as a server response
  if (valid) {
    // manually close the dialog
    dialog.close();
  } else {
    // don't close the dialog and let the user know there's an error
    // such as by setting the textfield's error message
    textfield.setCustomValidity('There was a problem');
    textfield.reportValidity();
  }
});
```

Reply:
Nice! will do! thanks for the example :) I'll update this